### DISCUSSION TITLE
TSP: Visit as many nodes as possible without exceeding distance d.

### DISCUSSION THREAD

**Discussion by BrandonWL** [google/or-tools#3393] [Routing (and legacy CP) questions] [answered]
I want to be able to visit as many clients as possible without the route distance exceeding distance d. I don't require every client gets traversed.

from this TSP example:
https://developers.google.com/optimization/routing/tsp#program1
I added a dimension to limit the route distance to 6000 and got no solution.

```py
    # Add Distance constraint.
    dimension_name = 'Distance'
    routing.AddDimension(
        transit_callback_index,
        0,  # no slack
        6000,  # vehicle maximum travel distance
        True,  # start cumul to zero
        dimension_name)
    distance_dimension = routing.GetDimensionOrDie(dimension_name)
    distance_dimension.SetGlobalSpanCostCoefficient(100)
```

From the VRP example:
https://developers.google.com/optimization/routing/vrp#entire_program1
I modified the dimension from 3000 -> 13000 and also got no solution
```py
    # Add Distance constraint.
    dimension_name = 'Distance'
    routing.AddDimension(
        transit_callback_index,
        0,  # no slack
        1300,  # vehicle maximum travel distance 3000 -> 1300
        True,  # start cumul to zero
        dimension_name)
    distance_dimension = routing.GetDimensionOrDie(dimension_name)
    distance_dimension.SetGlobalSpanCostCoefficient(100)
```

Is this even possible using OR tools? I've searched everywhere and couldn't find a solution.

**Comment by Mizux** [comment, marked as answer]
https://developers.google.com/optimization/routing/penalties

**Reply by BrandonWL** [reply]
Oops I guess I didn't search hard enough. Thanks Mizux!
